### PR TITLE
UI: Reformat .ui files with Qt Designer

### PR DESCRIPTION
**Pull request by derrod**
### Description

Reformats all `.ui` files with Qt Designer by opening and re-saving them to update them to the Qt 6 XML format and fix some formatting errors.

This also fixes `multiviewProjectorMenu` being declared in the wrong section.

### Motivation and Context

Every time we edit a `.ui` file with Qt Designer and save it we have to either manually undo these changes or manually stage only the hunks that contain the changes we want. That is silly.

(Qt will randomly change the size values in the file for some reason so we'll still have to skip those if we want to keep diffs down, but those are usually only one or two per file and that's much more managable. But I propose we just ignore those going forward since they're meaningless anyway.)

Replaces #8372 which got stuck in bikeshedding limbo, only covered the main file, and is conflicted.

### How Has This Been Tested?

OBS still compiles and looks the same.

### Types of changes

- Code cleanup (non-breaking change which makes code smaller or more readable)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by RytoEX** [comment]
What version of Qt Creator/Designer was used for this?

Looks like the macOS build didn't like the repeated use of "widget" as the name for some widgets.
image

I'm most of the way through checking the diff, but still have the two largest files to go.

**Comment by derrod** [comment]
> What version of Qt Creator/Designer was used for this?

Qt Designer 6.7.0 actually since that's what I had installed, didn't seem to have any changes from 6.6.2 though.

Not sure why it did the double `widget` thing, might have to manually fix that.

File: UI/forms/OBSBasic.ui
Comment (Warchamp7) [inline UI/forms/OBSBasic.ui:179]: Need someone on Mac to test if this causes issues. I remember a long time ago there was some horrific issue on Mac caused by Qt adding this attribute that was mitigated by removing it.

Comment (RytoEX) [reply UI/forms/OBSBasic.ui:179]: You might be thinking of Linux? I didn't find anything at a glance for macOS.

https://github.com/obsproject/obs-studio/commit/9ee7547e192fac25cf08124d57037cb124b69c38

File: UI/forms/OBSPermissions.ui
Comment (RytoEX) [inline UI/forms/OBSPermissions.ui:277]: Why even are there blank nameless widgets here? Layout reasons? @gxalpha ?

I feel like there's a mechanism for "add an element to the second column in a layout" without having to spawn an empty widget. Or else a spacer item should work instead of a full widget? If so, that change would be out of scope for this PR, but we could change it in a separate PR.

If not, then yeah, they'll need unique names, which again, we could probably fix in a separate PR to keep the scope/intent of this PR constrained.

**Review by RytoEX — COMMENTED** [review]
Musing thoughts while trying to move this forward.

File: UI/forms/OBSPermissions.ui
Comment (gxalpha) [reply UI/forms/OBSPermissions.ui:277]: Would need to dig in to see what was going on there. I mostly took the layout from an earlier attempt at this dialog (by Warchamp and Matt) and then modified it, it’s quite possible that I copied this part and made bad modifications here.

File: UI/forms/OBSBasic.ui
Comment (RytoEX) [reply UI/forms/OBSBasic.ui:179]: cc @PatTheMav @gxalpha I don't recall if we've ever had issues with native widgets on macOS?
cc @GeorgesStavracas See above.

#3782 did previously remove `native="true"` from this widget.
https://github.com/obsproject/obs-studio/commit/9ee7547e192fac25cf08124d57037cb124b69c38#diff-c4723e7657a9d380fa8407cf136dc7e888e73120f82b0552722d45a709f7120fL170-R170

**Review by RytoEX — CHANGES_REQUESTED** [review]
I've reviewed every file. The only requested changes are the ones in already opened conversations.

File: UI/forms/OBSBasic.ui
Comment (GeorgesStavracas) [reply UI/forms/OBSBasic.ui:179]: These native surfaces were causing trouble with Wayland back in the day. Not sure how that goes now.

Comment (RytoEX) [reply UI/forms/OBSBasic.ui:179]: Are we feeling adventurous to re-examine this? The way I see it, we have a few paths forward:
1. Test this on Wayland before merging.
2. Revert just these changes for now. Test the `native="true"` changes on Wayland later. Open a new PR for those later if testing goes well.
3. Merge it and be on the lookout for issues. Revert the `native="true"` changes if needed.

I would like this to not get blocked for a prolonged period of time.

File: UI/forms/OBSPermissions.ui
Comment (RytoEX) [reply UI/forms/OBSPermissions.ui:277]: See #10473

**Comment by RytoEX** [comment]
I break, I fix. As mentioned in #10473, `UI/forms/OBSPermissions.ui` will probably need a second pass from Qt Designer/Creator on your end to pick up the enum/namespace changes. For some reason, when I processed it in Qt Creator 12, it did not catch those.

File: UI/forms/OBSPermissions.ui
Comment (RytoEX) [reply UI/forms/OBSPermissions.ui]: Should be resolved now.

File: UI/forms/OBSBasic.ui
Comment (GeorgesStavracas) [reply UI/forms/OBSBasic.ui:179]: My personal preference is (2) but it's more laborious than the others. If it's too much trouble, just merge with it.

Comment (RytoEX) [reply UI/forms/OBSBasic.ui:179]: Hmm. Thanks. Option 2 is probably "safest", but does also require _someone_ to go test on Wayland and check for issues and then PR/merge those changes at a later date, which is currently not in my capacity to do so. @derrod ?

**Comment by RytoEX** [comment]
> I break, I fix. Merging #10473 first caused a small merge conflict.
> 
> As mentioned in #10473, `UI/forms/OBSPermissions.ui` will probably need a second pass from Qt Designer/Creator on your end to pick up the enum/namespace changes. For some reason, when I processed it in Qt Creator 12, it did not catch those.

We discussed this off-thread and discovered the differences were down to different versions of Qt Designer (`bin\designer.exe`) included with different versions of Qt. Qt 6.7.0's designer.exe produces the enum differences seen here. This is due to changes in Qt 6.7.0, but these changes _should_ be backwards compatible, and CI seems able to compile with these changes, all the way back to Qt 6.2.4 on Ubuntu 22.04.
https://bugreports.qt.io/browse/QTBUG-118473

We'll still need to decide on the native widgets issue:
https://github.com/obsproject/obs-studio/pull/10419#discussion_r1538215097

File: UI/forms/OBSBasic.ui
Comment (tytan652) [reply UI/forms/OBSBasic.ui:179]: Looks fine on GNOME Wayland after a quick test resizing arround with Studio mode on and off.

**Comment by RytoEX** [comment]
Final blockers:
1. Choose which version of designer.exe or Creator to use for formatting.
2. Decide on the wayland native widgets problem.

The main challenge with Creator is that it's not delivered in-sync with Qt versions. Qt Creator 13 based on Qt 6.6.0 was released on April 3, 2024, but Qt 6.7.0 was released on April 2, 2024. There's no guarantee that the version of Creator you launch is using the version of Qt (designer) you have installed, as far as I can tell.

File: UI/forms/OBSBasic.ui
Comment (RytoEX) [reply UI/forms/OBSBasic.ui:179]: Sounds like we'll give this a scream test then.

**Comment by RytoEX** [comment]
> Final blockers:
>     1. Choose which version of designer.exe or Creator to use for formatting.
>     2. Decide on the wayland native widgets problem.
> 
> The main challenge with Creator is that it's not delivered in-sync with Qt versions. Qt Creator 13 based on Qt 6.6.0 was released on April 3, 2024, but Qt 6.7.0 was released on April 2, 2024. There's no guarantee that the version of Creator you launch is using the version of Qt (designer) you have installed, as far as I can tell.

I think:
1. We should use the version of designer.exe that matches the version of Qt we ship on Windows and macOS. Right now that is 6.6.2. When we update those platforms, we can update the .ui files in tandem. Unless @Warchamp7 has a different opinion?
2. We're going to go with these changes. We'll do a scream test and be vigilant for reports of breakage.

**Comment by Warchamp7** [comment]
> > Final blockers:
> > 1. Choose which version of designer.exe or Creator to use for formatting.
> > 2. Decide on the wayland native widgets problem.
> > The main challenge with Creator is that it's not delivered in-sync with Qt versions. Qt Creator 13 based on Qt 6.6.0 was released on April 3, 2024, but Qt 6.7.0 was released on April 2, 2024. There's no guarantee that the version of Creator you launch is using the version of Qt (designer) you have installed, as far as I can tell.
> 
> I think:
> 
> 1. We should use the version of designer.exe that matches the version of Qt we ship on Windows and macOS. Right now that is 6.6.2. When we update those platforms, we can update the .ui files in tandem. Unless @Warchamp7 has a different opinion?
> 2. We're going to go with these changes. We'll do a scream test and be vigilant for reports of breakage.

Stamp of approval from me.

**Comment by derrod** [comment]
> 1. We should use the version of designer.exe that matches the version of Qt we ship on Windows and macOS. Right now that is 6.6.2. When we update those platforms, we can update the .ui files in tandem. Unless @Warchamp7 has a different opinion?

Could just delay until the update happens, otherwise we'd have two commits that touch many `.ui` files that make blaming even more difficult.

**Comment by RytoEX** [comment]
> > 1. We should use the version of designer.exe that matches the version of Qt we ship on Windows and macOS. Right now that is 6.6.2. When we update those platforms, we can update the .ui files in tandem. Unless @Warchamp7 has a different opinion?
> 
> Could just delay until the update happens, otherwise we'd have two commits that touch many `.ui` files that make blaming even more difficult.

Sure. I'll try to spin dependencies updates soon.

**Comment by RytoEX** [comment]
Per off-thread discussion with @Warchamp7 , we've re-run the .ui files through Qt 6.6.2's designer.exe for now to unblock this and fix the worst issues. Finishing up some local test builds.

**Review by RytoEX — APPROVED** [review]
Re-tested this locally on Windows 11 and it seems fine. This no longer includes the Qt 6.7 changes for fully-qualified enums, which is supposedly only compatible back to Qt 6.5.

**Pull request merged**
